packages/ajax/test/AjaxClass.interceptors-cache.test.js

```diff
@@ -0,0 +1,713 @@
+      useCache: 'always',
+      timeToLive: 1000,
+    });
+
+    server.respondWith(/^\/posts$/, xhr => {
+      xhr.respond(200, { 'Content-Type': 'application/json' }, '{}');
+    });
+
+    server.respondWith(/^\/posts\/1$/, xhr => {
+      xhr.respond(200, { 'Content-Type': 'application/json' }, '{}');
+    });
+
+    const actionConfig = {
+      lionCacheOptions: {
+        invalidateUrlsRegex: /posts/gi,
+      },
+    };
+
+    return ajax
+      .get('/test', actionConfig)
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => ajax.get('/posts'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/posts'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.get('/posts/1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.get('/posts/1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(3);
+      })
+      .then(() => ajax.post('/test', {}, actionConfig))
+      .then(() => {
+        expect(server.requests.length).to.equal(4);
+      })
+      .then(() => ajax.get('/posts'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .then(() => ajax.get('/posts/1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .finally(() => {
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('does not returns the cached object on second call default timeToLive`', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 0,
+    });
+
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+
+    return ajax
+      .get('/test')
+      .then(() => {
+        const clock = sinon.useFakeTimers();
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test')).to.be.true;
+        clock.tick(1);
+        clock.restore();
+      })
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it("returns the cached object on second call with `useCache: 'always'`, with querystring parameters", () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 100,
+    });
+
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+
+    return ajax
+      .get('/test', {
+        params: {
+          q: 'test',
+          page: 1,
+        },
+      })
+      .then(() => {
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test')).to.be.true;
+      })
+      .then(() =>
+        ajax.get('/test', {
```

Comment:
Shall we add a test with different params, to see if the server requests length is then 2?